### PR TITLE
ignore timeout error in job updater

### LINKED ISSUE #8545: JobUpdater crashing with timeout
It happened to the CMS@Home agent, we should just skip the cycle (and maybe register it in logdb).

Traceback was
```
2018-04-05 10:50:44,217:140498145695488:WARNING:Requests:Http request failed, retrying once again..
2018-04-05 11:10:44,399:140498145695488:ERROR:JobUpdaterPoller:Caught unexpected exception in JobUpdater: timed out
Traceback (most recent call last):
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.10.patch2/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 84, in algorithm
    self.synchronizeJobPriority()
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.10.patch2/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 116, in synchronizeJobPriority
    workflowsToCheck = [x for x in self.workqueue.getAvailableWorkflows()]
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.10.patch2/lib/python2.7/site-packages/WMCore/Services/WorkQueue/WorkQueue.py", line 193, in getAvailableWorkflows
    {'reduce': False, 'stale': 'update_after'})
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.10.patch2/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 527, in loadView
    (self.name, design, view), encodedOptions)
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.10.patch2/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 117, in get
    encode, decode, contentType)
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.10.patch2/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 120, in makeRequest
    encode, decode, contentType)
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.10.patch2/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 153, in makeRequest
    encoder, decoder, contentType)
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.10.patch2/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 272, in makeRequest_httplib
    response, result = conn.request(uri, method=verb, body=encoded_data, headers=headers)
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/external/py2-httplib2/0.7.1-comp/lib/python2.7/site-packages/httplib2/__init__.py", line 1436, in request
    (response, content) = self._request(conn, authority, uri, request_uri, method, body, headers, redirections, cachekey)
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/external/py2-httplib2/0.7.1-comp/lib/python2.7/site-packages/httplib2/__init__.py", line 1188, in _request
    (response, content) = self._conn_request(conn, request_uri, method, body, headers)
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/external/py2-httplib2/0.7.1-comp/lib/python2.7/site-packages/httplib2/__init__.py", line 1158, in _conn_request
    response = conn.getresponse()
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/external/python/2.7.13/lib/python2.7/httplib.py", line 1121, in getresponse
    response.begin()
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/external/python/2.7.13/lib/python2.7/httplib.py", line 438, in begin
    version, status, reason = self._read_status()
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/external/python/2.7.13/lib/python2.7/httplib.py", line 394, in _read_status
    line = self.fp.readline(_MAXLINE + 1)
  File "/data/srv/wmagent/v1.1.10.patch2/sw/slc7_amd64_gcc630/external/python/2.7.13/lib/python2.7/socket.py", line 480, in readline
    data = self._sock.recv(self._rbufsize)
timeout: timed out
2018-04-05 11:17:10,875:140498145695488:ERROR:BaseWorkerThread:Error in worker algorithm (1):
Backtrace:
  <WMComponent.JobUpdater.JobUpdaterPoller.JobUpdaterPoller object at 0x7fc84d6857d0> <@========== WMException Start ==========@>
Exception Class: JobUpdaterException
Message: Caught unexpected exception in JobUpdater: timed out
```

### PR DESCRIPTION
**Pull request by ticoann**
fixes #8545

### CODE REVIEW

File: src/python/WMComponent/JobUpdater/JobUpdaterPoller.py [hunk 1]
```diff
@@ -93,11 +93,19 @@ def algorithm(self, parameters=None):
             msg += "transactions postponed until the next polling cycle\n"
             msg += str(ex)
             logging.exception(msg)
+        except CouchError as ex:
+            if ex.status is None and ex.reason is None:
```
Comment (amaltaro) [inline src/python/WMComponent/JobUpdater/JobUpdaterPoller.py]: Looking at the code, I wonder if we can ever have ex.reason=None? It seems reason would always have a non None value (unless we explicitly construct it with None)

**Review by amaltaro — COMMENTED** [review]
Patch looks good, Seangchan.
There is only one thing that's not clear to me, I made a comment in the code

File: src/python/WMComponent/JobUpdater/JobUpdaterPoller.py (hunk 1 above)
Comment (ticoann) [reply src/python/WMComponent/JobUpdater/JobUpdaterPoller.py]: Thanks for the review, Alan.
I am not sure but this particular error contains reason is None

```
2018-04-14 09:24:41,838:139750312437504:ERROR:JobUpdaterPoller:Caught unexpected exception in JobUpdater: CouchError - reason: None, data: {'stale': 'update_after', 'reduce': 'false'} result: None
Traceback (most recent call last):
```
I just don't want to be too general for catching errors at least for now.

**Comment by ticoann** [comment]
OK finally it failed again and handled properly.
```
2018-05-04 18:19:44,724:139740403103488:ERROR:JobUpdaterPoller:Couch Server error occured. Mostly due to time out
CouchError - reason: None, data: {'stale': 'update_after', 'reduce': 'false'} result: None
Traceback (most recent call last):
  File "/data/srv/wmagent/v1.1.12.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.12.patch2/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 84, in algorithm
    self.synchronizeJobPriority()
  File "/data/srv/wmagent/v1.1.12.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.12.patch2/lib/python2.7/site-packages/WMComponent/JobUpdater/JobUpdaterPoller.py", line 124, in synchronizeJobPriority
    workflowsToCheck = [x for x in self.workqueue.getAvailableWorkflows()]
  File "/data/srv/wmagent/v1.1.12.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.12.patch2/lib/python2.7/site-packages/WMCore/Services/WorkQueue/WorkQueue.py", line 193, in getAvailableWorkflows
    {'reduce': False, 'stale': 'update_after'})
  File "/data/srv/wmagent/v1.1.12.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.12.patch2/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 527, in loadView
    (self.name, design, view), encodedOptions)
  File "/data/srv/wmagent/v1.1.12.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.12.patch2/lib/python2.7/site-packages/WMCore/Services/Requests.py", line 117, in get
    encode, decode, contentType)
  File "/data/srv/wmagent/v1.1.12.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.12.patch2/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 123, in makeRequest
    getattr(e, "reason", None), data)
  File "/data/srv/wmagent/v1.1.12.patch2/sw/slc7_amd64_gcc630/cms/wmagent/1.1.12.patch2/lib/python2.7/site-packages/WMCore/Database/CMSCouch.py", line 155, in checkForCouchError
    raise CouchError(reason, data, result, status)
CouchError: CouchError - reason: None, data: {'stale': 'update_after', 'reduce': 'false'} result: None
2018-05-04 18:20:18,821:139740403103488:INFO:BaseWorkerThread:JobUpdaterPoller took 320.977 secs to execute
```

Merging it.

**Pull request merged**